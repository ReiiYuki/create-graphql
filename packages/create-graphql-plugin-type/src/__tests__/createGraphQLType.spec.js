import path from 'path';
import typePlugin from '../index';

it('should generate a type template from fields', async () => {
  const fields = [
    {
      name: 'name',
      description: 'user name',
      required: true,
      originalType: 'String',
      resolve: 'obj.name',
      type: 'GraphQLString',
      flowType: 'string',
    },
    {
      name: 'password',
      description: 'hashed user password',
      required: false,
      originalType: 'String',
      resolve: 'obj.password',
      type: 'GraphQLString',
      flowType: 'string',
    },
    {
      name: 'email',
      description: 'user email',
      required: false,
      originalType: 'String',
      resolve: 'obj.email',
      type: 'GraphQLString',
      flowType: 'string',
    },
    {
      name: 'active',
      required: false,
      originalType: 'Boolean',
      resolve: 'obj.active',
      type: 'GraphQLBoolean',
      flowType: 'boolean',
    },
  ];

  const typeName = 'Example';
  const plugin = typePlugin({ name: typeName });
  const templates = plugin.generateTemplates({ fields });

  expect(templates).toMatchSnapshot();
});
