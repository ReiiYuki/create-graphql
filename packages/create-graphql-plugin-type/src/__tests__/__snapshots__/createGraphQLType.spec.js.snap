// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate a type template from fields 1`] = `
"export default new GraphQLObjectType({
  name: \\"Example\\",
  description: 'Represents NAME',
  fields: () => ({
    id: globalIdField('NAME'),
    example: {
      type: GraphQLString,
      description: 'My example field',
      resolve: obj => obj.example
    }
  }),
  interfaces: () => [NodeInterface]
});"
`;
